todo_app/app.py

```diff
@@ -1,33 +1,100 @@
 from flask import Flask, render_template, request
+from flask_login import LoginManager, current_user, login_required, login_user
+import os
+import requests
 from werkzeug.utils import redirect
 from todo_app.data.mongodb_items import get_items, create_item, delete_item, change_item_status
 from todo_app.flask_config import Config
+from todo_app.models.User import User
 from todo_app.models.ViewModel import ViewModel
+from functools import wraps
+from flask import abort
 
 def create_app():
     app = Flask(__name__)
     app.config.from_object(Config())
+    login_manager = LoginManager()
+    app.config['LOGIN_DISABLED'] = os.getenv('LOGIN_DISABLED') == 'True'
+
+    auth_mapping = {"78739678": "writer"}
+
+    def user_in_group(current_user, group):
+        if(app.config.get('LOGIN_DISABLED') == True):
+            return True
+        if current_user is None or not current_user.is_authenticated:
+            return False
+        if auth_mapping[current_user.get_id()] != group:
+            return False
+        return True
+
+    def writer_required(f):
+        @wraps(f)
+        def decorated_function(*args, **kwargs):
+            if not user_in_group(current_user, "writer"):
+                abort(404)
```

Comment:
I quite like the use of 404 here - 401/403 can feel more appropriate but refusing to talk about what exists to those who shouldn't have permission is a very sensible policy - it prompted me to think a bit and [I liked this article](https://www.bennadel.com/blog/2400-handling-forbidden-restful-requests-401-vs-403-vs-404.htm#:~:text=The%20three%20status%20codes%20that,404%20%2D%20Not%20Found) I found